### PR TITLE
Prevent click on save button while data is saving

### PR DESCRIPTION
**Pull request by YaroslavOvdii**
@sofiiakvasnevska 
## Issue
https://github.com/Fliplet/fliplet-studio/issues/5360

## Description
Prevent click on save button while data is saving

## Screenshots/screencasts
click-while-save

## Backward compatibility

This change is fully backward compatible.

## Notes
This solution work in pair with this [PR.](https://github.com/Fliplet/fliplet-widget-image-editor/pull/21)

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,3 +1,4 @@
 # IDE and OS files
 .DS_Store
 .idea
+.history
```
Comment (tonytlwu) [inline .gitignore]: @squallstar Can you confirm if we want to ignore `.history` file?

```suggestion
```

Comment (squallstar) [reply .gitignore]: @tonytlwu yes it's fine to add `.history` to `.gitignore`, we discussed this in the past, it's a feature on VSCode which basically mimics `.git`

File: js/interface.js [hunk 2]
```diff
@@ -65,11 +65,13 @@ function attahObservers() {
 
   // 1. Fired from Fliplet Studio when the external save button is clicked
   Fliplet.Widget.onSaveRequest(function() {
+    Fliplet.Widget.toggleSaveButton(false);
```
Comment (tonytlwu) [inline js/interface.js]:
```suggestion
    Fliplet.Widget.toggleSaveButton(false);

```

File: js/interface.js [hunk 3]
```diff
@@ -65,11 +65,13 @@ function attahObservers() {
...
     if (linkActionProvider && !$('#pinch').is(':checked') && !$('#none').is(':checked')) {
       return linkActionProvider.forwardSaveRequest();
     } else {
       forwardSaveRequestFilePicker();
     }
+    Fliplet.Widget.complete();
```
Comment (tonytlwu) [inline js/interface.js]:
```suggestion

    Fliplet.Widget.complete();
```

**Pull request merged**